client/java/src/test/java/io/openlineage/client/transports/HttpTransportTest.java

```diff
@@ -269,6 +275,52 @@ void customHeaders() throws IOException {
         .containsEntry("testHeader2", "test2");
   }
 
+  @Test
+  void gzipCompression() throws IOException {
+    HttpConfig config = new HttpConfig();
+    config.setUrl(URI.create("https://localhost:1500/api/v1/lineage"));
+    config.setCompression(HttpConfig.Compression.GZIP);
+
+    CloseableHttpClient http = mock(CloseableHttpClient.class);
+    Transport transport = new HttpTransport(http, config);
+    OpenLineageClient client = new OpenLineageClient(transport);
+
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class, RETURNS_DEEP_STUBS);
+    when(response.getStatusLine().getStatusCode()).thenReturn(200);
+    when(response.getEntity().isStreaming()).thenReturn(true);
+    Map<String, HttpPost> map = new HashMap<>();
+    when(http.execute(any(HttpUriRequest.class)))
+        .thenAnswer(
+            invocation -> {
+              map.put("test", invocation.getArgument(0));
+              return response;
+            });
+
+    client.emit(runEvent());
+
+    HttpEntity resultEntity = map.get("test").getEntity();
+    assertThat(resultEntity).isInstanceOf(GzipCompressingEntity.class);
+
+    GzipCompressingEntity gzippedEntity = (GzipCompressingEntity) resultEntity;
+    ByteArrayOutputStream compressedData = new ByteArrayOutputStream();
+    gzippedEntity.writeTo(compressedData);
+
+    GZIPInputStream uncompressedStream =
```

Comment:
I like this test 👍

Reply:
Well, I'm not really. The only thing is tested here is ApacheHTTP client classes